lib/model/message.dart

```diff
@@ -93,32 +93,42 @@ class MessageStoreImpl with MessageStore {
   }
 
   void handleUpdateMessageEvent(UpdateMessageEvent event) {
+    _handleUpdateMessageEventTimestamp(event);
     _handleUpdateMessageEventContent(event);
     // TODO(#150): Handle message moves.  The views' recipient headers
     //   may need updating, and consequently showSender too.
+    for (final view in _messageListViews) {
+      view.notifyListenersIfAnyMessagePresent(event.messageIds);
+    }
   }
```

Comment:
```
message: Update all edit timestamps on UpdateMessageEvent
```

So in this commit, message lists check `event.messageIds` to see if they need to notify listeners, and they don't check `event.messageId` for that purpose. I think the API doc isn't quite explicit that `message_ids` will include the message ID for a pure content edit: https://zulip.com/api/get-events#update_message

I might be splitting hairs, though. On the `message_id` field, the doc does have this:

> […] the separate `message_ids` field, which is guaranteed to include `message_id`.

…but then again, here's the full sentence that quote is from:

> If the channel or topic was changed, the set of moved messages is encoded in the separate `message_ids` field, which is guaranteed to include `message_id`.

I did just test a pure content edit, though, and the message ID did appear in `message_ids`.

Reply:
It does appear to be the case that `message_ids` [always show up](https://github.com/zulip/zulip/blob/be9369541af01557f7ccd35d655324320b30e6be/zerver/actions/message_edit.py#L388).

However, this feels like one of the cases where it will be more convenient if we know that the field should never be missing or nullable. Having our OpenAPI spec [properly defined](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/Outdated.20API.20response.20doc.20in.20get_streams/near/1390715) so that a "only present if ..." field is marked as nullable/required explicitly will definitely help.

Reply:
Ah, so I'm not actually looking at whether the `message_ids` field is present or not; I'm assuming it is present. I'm looking for an API guarantee that, in a pure content-edit event, `message_ids` will include the ID of the edited message, instead of being an empty array. If it can be an empty array, then I expect a message list would fail to show the new content immediately, because we wouldn't be calling `MessageListView.notifyListeners`).

Reply:
Hmm, I agree it'd be good to have that better nailed down. We have this in our API types in zulip-mobile:
```js
  // Any stream/topic changes apply to all of message_ids, which is
  //   guaranteed to include message_id.
  // TODO(server-future): It'd be nice for these to be sorted; NB they may
  //   not be.  As of server 5.0-dev-3868-gca17a452f (2022-02), there's no
  //   guarantee of that in the API nor, apparently, in the implementation.
  message_ids: $ReadOnlyArray<number>,
```

Looks like that dates to zulip/zulip-mobile@4ce9e20b5ba1eba0f560506c1cc835fce1834419 . That's actually after the era where we'd started having good API docs outside the zulip-mobile type definitions. But I don't see in that PR thread https://github.com/zulip/zulip-mobile/pull/5155 , or the [chat thread](chat thread) it links to, any discussion of this particular question.

Anyway, I do think we intend to make that guarantee. I'll ask in the thread and see about making it explicit in the API docs.